Use a variable to determine if a type should be added or not when resolved.

dpomerantz-jci opened the pull request:
### All Submissions:

- [X] Have you followed the guidelines in our [Contributing](https://github.com/lukeautry/tsoa/tree/master/docs/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/lukeautry/tsoa/pulls) for the same update/change?
- [ ] Have you written unit tests?
- [ ] Have you written unit tests that cover the negative cases (i.e.: if bad data is submitted, does the library respond properly)?
- [X] This PR is associated with an existing issue?

**Closing issues**

closes #1330 

### If this is a new feature submission:

- [ ] Has the issue had a maintainer respond to the issue and clarify that the feature is something that aligns with the [goals](https://github.com/lukeautry/tsoa#goal) and [philosophy](https://github.com/lukeautry/tsoa#philosophy) of the project?

**Potential Problems With The Approach**

<!-- if there are areas of the solution that you think might have limitations or potential "gotchas", then discuss them here -->
<!-- if you have many questions, then please consider discussing them in the issue thread (unless you need to share code to illustrate the questions) -->

**Test plan**

<!-- explain why the unit tests that you added are demonstrating that the feature works -->

Negative test was added, though it is relatively brittle.  The goal of this change is to ensure that heritage types for interfaces do not get added to the refType map simply because they are heritage types.  To test this, first a new model was created, HeritageTestModel2.  This was done so that any changes changes to or expectations upon HeritageTestModel will not affect this test.  HeritageTestModel extends HeritageBaseModel. In the fixture that includes HeritageTestModel, a test was added to ensure that HeritageTestModel2 is created, but HeritageBaseModel is not.  This can be broken if HeritageTestModel2's heritage is changed, and the the test will still pass.  It can also be broken be adding HeritageBaseModel to the model so that it is included in the refType map.  This would cause the test to fail even though it is doing what it is supposed to do.  It was verified that by adding a type which is in the refType map, the test fails.

Another test was added so that if HeritageTestModel2 is never found, it alerts the tester that this was a false positive.

WoH reviewed (commented):
Notes:
- `add` is not a very descriptive name, `addToRefTypeMap` would be better.
- If we don't add models added by heritage clauses to ref types, we should make sure they don't end up in the spec. Aka add a small test case.
- I'm pretty sure we don't need them at runtime, since we combine the props into the underlying interface, so while we don't need an integration test, it wouldn't hurt either.

dpomerantz-jci commented:
> Notes:
> 
> * `add` is not a very descriptive name, `addToRefTypeMap` would be better.
> * If we don't add models added by heritage clauses to ref types, we should make sure they don't end up in the spec. Aka add a small test case.
> * I'm pretty sure we don't need them at runtime, since we combine the props into the underlying interface, so while we don't need an integration test, it wouldn't hurt either.

Parameter renamed as requested.

I've added a test, and hope this is sufficient.

dpomerantz-jci commented:
@WoH Tests were failing because lint changed things on commit.  I reverted the changes and added the disable comment.  Tests should now pass.  Please lmk if there are further changes required, or otherwise approve and merge.  Thanks.

WoH commented:
https://github.com/lukeautry/tsoa/releases/tag/v5.0.0-alpha.3

WoH commented:
Current main branch is the 5.x
Main reason was to drop support for older node versions and be safe for some potentially breaking fixes.
